pkg/collector/corechecks/snmp/devicecheck/devicecheck.go

```diff
@@ -76,7 +76,12 @@ func (d *DeviceCheck) GetIDTags() []string {
 // GetDeviceHostname returns DeviceID as hostname if UseDeviceIDAsHostname is true
 func (d *DeviceCheck) GetDeviceHostname() string {
 	if d.config.UseDeviceIDAsHostname {
-		return deviceHostnamePrefix + d.config.DeviceID
+		hostname := deviceHostnamePrefix + d.config.DeviceID
+		normalizedHostname := common.NormalizeHost(hostname)
+		if hostname != normalizedHostname {
+			log.Warnf("Invalid namespace, device hostname %s normalized to: %s", hostname, normalizedHostname)
+		}
+		return normalizedHostname
```

Comment:
If the normalisation fail, I think it might be preferred for the integration run to report an error than continuing with an empty hostname. It might be hard to debug similar to @FlorianVeaux comment here https://github.com/DataDog/datadog-agent/pull/10702/files/391fff8147764c2b28f8b16b9d53bc3aa0cfc9d5#r794588215

Reply:
Empty hostname here would end up in using agent's host, thus disabling `use_device_id_as_hostname`. I guess what could help here is an additional normalization / validation function restricted to namespace. It should respect hostname policy even if it is not used as a hostname